python/cuspatial/cuspatial/core/interpolate.py

```diff
@@ -0,0 +1,182 @@
+
+    cuspatial massively outperforms scipy however when many
+    splines are fit simultaneously. Data must be arranged in a SoA format,
+    and the inclusive/exclusive prefix_sum of the separate curves must also
+    be passed to the function.
+
+        t = cudf.Series(np.repeat(np.arange(100), 1000)).astype('float32')
+        y = cudf.Series(np.random.random(100*1000)).astype('float32')
+        prefix_sum = cudf.Series(np.arange(1000)*100).astype('int32')
+        new_samples = cudf.Series(np.repeat(np.linspace(0, 100, 1000), 1000)
+            .astype('float32'))
+
+        curve = cuspatial.CubicSpline(t, y, prefixes=prefix_sum)
+        new_points = curve(new_samples, prefix_sum*10)
+
+    """
+
+    def __init__(self, t, y, ids=None, size=None, prefixes=None):
+        """
+        Computes various error preconditions on the input data, then
+        calls C++/Thrust code to compute cubic splines for each set of input
+        coordinates in parallel.
+        """
+        # error protections:
+        if len(t) < 5:
+            raise ValueError(
+                "Use of GPU cubic spline requires splines of length > 4"
+            )
+        if not isinstance(t, Series):
+            raise TypeError(
+                "Error: input independent vars must be cudf Series"
+            )
+        if not isinstance(y, (Series, DataFrame)):
+            raise TypeError(
+                "Error: input dependent vars must be cudf Series or DataFrame"
+            )
+        if not len(t) == len(y):
+            raise TypeError(
+                "Error: dependent and independent vars have different length"
+            )
+        if ids is None:
+            self.ids = Series([0, 0]).astype("int32")
+        else:
+            if not isinstance(ids, Series):
+                raise TypeError("cuspatial.CubicSpline requires a cudf.Series")
+            if not ids.dtype == np.int32:
+                raise TypeError("Error: int32 only supported at this time.")
+            self.ids = ids
+        self.size = size if size is not None else len(t)
+        if not isinstance(self.size, int):
+            raise TypeError("Error: size must be an integer")
+        if not ((len(t) % self.size) == 0):
+            raise ValueError(
+                "Error: length of input is not a multiple of size"
+            )
+        if not isinstance(t, Series):
+            raise TypeError("cuspatial.CubicSpline requires a cudf.Series")
+        if not t.dtype == np.float32:
+            raise TypeError("Error: float32 only supported at this time.")
+        if not isinstance(y, Series):
+            raise TypeError("cuspatial.CubicSpline requires a cudf.Series")
+        if not y.dtype == np.float32:
+            raise TypeError("Error: float32 only supported at this time.")
+        self.t = t
+        self.y = y
+        if prefixes is None:
+            self.prefix = Series(
+                np.arange((len(t) / self.size) + 1) * self.size
```

Comment:
You should use `cupy.arange` here instead to do this on the GPU

Reply:
:thumbsup: